sdk/main/src/Transaction.cc

```diff
@@ -763,9 +794,7 @@ Transaction<SdkRequestType>::Transaction(
 
   mImpl->mTransactionMemo = mImpl->mSourceTransactionBody.memo();
 
-  // This constructor is only used in fromBytes(), which means a Transaction was frozen and serialized using toBytes().
-  // As such, this Transaction should be constructed as frozen.
-  mImpl->mIsFrozen = true;
+  mImpl->mIsFrozen = shouldFreeze;
 }
```

Comment:
I'm a bit confused by the added logic in this function. If a `proto::Transaction` doesn't contain signed body bytes, that means the `Transaction` shouldn't be frozen? But will this ever happen? If you use `toBytes` will that ever create an empty `proto::Transaction`? It seems like the number of signatures added to the `proto::Transaction` should be looked at here somewhere, it was my understanding that if a signature has been added to a `proto::Transaction` then it should be frozen, otherwise it shouldn't.

Reply:
Changed the logic for shouldFreeze now it freezes on sigmap empty.